info-ui/src/main/java/com/example/info_ui/AdditionalInfoScreenFragment.kt

```diff
@@ -1,70 +1,275 @@
 
     private val viewModel by viewModels<AdditionalInfoScreenViewModel>()
+    private val args: AdditionalInfoScreenFragmentArgs by navArgs()
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        binding = DataBindingUtil.inflate(
-            inflater, R.layout.fragment_additional_info_screen, container, false
-        )
-        binding.lifecycleOwner = this
-        binding.viewModel = viewModel
-
-        viewLifecycleOwner.lifecycleScope.launch {
-            viewModel.events.collect { handleEvent(it) }
+        val days = args.days
+
+        return ComposeView(requireContext()).apply {
+            setContent {
+                AppTheme {
+                    Surface(color = MaterialTheme.colorScheme.background) {
+                        ContentView(days = days)
+                    }
+                }
+            }
         }
+    }
 
-        val day = day
+    @OptIn(ExperimentalPagerApi::class)
+    @Composable
+    fun ContentView(
+        days: Array<String>
+    ) {
+        val coroutineScope = rememberCoroutineScope()
+        val pagerState = rememberPagerState()
 
-        viewModel.forecast.observe(viewLifecycleOwner) { data ->
+        Column(modifier = Modifier.fillMaxSize()) {
 
-            val forecastList = data?.list?.filter { it -> it.date.contains(day) }.orEmpty()
-            val item = forecastList.first().dateLong * 1000
-            val sdf = SimpleDateFormat("EEEE, d MMMM")
-            val dateAndTime = Date(item)
-            val dateAndTimeFormat = sdf.format(dateAndTime)
-            viewModel.dayValue.value = dateAndTimeFormat
-            setupRecyclerView(requireContext(), forecastList)
+            Button(
+                onClick = {
+                    activity?.onBackPressed()
+                },
+                modifier = Modifier
+                    .wrapContentSize()
+                    .padding(top = 8.dp, start = 8.dp)
+            ) {
+                Image(
+                    painter = painterResource(id = R.drawable.ic_back_button),
+                    contentDescription = null,
+                )
+            }
+
+            PagerContent(
+                pagerState = pagerState,
+                days = days,
+                scope = coroutineScope
+            )
         }
+    }
 
-        return binding.root
+    @SuppressLint("CoroutineCreationDuringComposition")
+    @OptIn(ExperimentalPagerApi::class)
+    @Composable
+    fun PagerContent(
+        pagerState: PagerState,
+        days: Array<String>,
+        scope: CoroutineScope
+    ) {
+        var isPageScrolled by remember { mutableStateOf(false) }
+        val day = args.day.removeRange(2, 12)
+        val item = days.find { it.contains(day) }
+        val index = days.indexOf(item)
+        Column(modifier = Modifier.fillMaxSize()) {
+
+            HorizontalPager(
+                count = 5,
```

Comment:
make this count dynamic value instead of 5. what if backend returns 3 days or 7?